File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.js
function solution(n, stations, w) {
    let answer = 0;
    let curIdx = 1;
    let stIdx = 0;
    while(curIdx <= n) {
        if(curIdx >= stations[stIdx] - w) {
            curIdx = stations[stIdx] + w + 1;
            stIdx++;
        } else {
            answer++;
            curIdx += 2 * w + 1;
        }
    }
    return answer;
}

File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.py
def solution(n, stations, w):
    answer = 0
    cur_loc = 1
    idx = 0
    
    while cur_loc <= n:
        if idx < len(stations) and cur_loc >= stations[idx] - w:
            cur_loc = stations[idx] + w + 1
            idx += 1
        else:
            cur_loc += 2 * w + 1
            answer += 1
            
    return answer
            